### PR TITLE
Store layout info

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-8468](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8468) Display Search Result Items in Compact View Layout

This PR addresses 2 issues:
1) Layout button should be clickable not only on the title and layout icon, but also on the background.
2) Layout should be stored in settings and loads every session.

### Declarations

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.

### Reviewers

@ramramps 

### FYIs

@Racel

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Search/LibraryContainerView.xaml [hunk 1]
```diff
@@ -382,16 +382,16 @@
             <ContentControl Style="{StaticResource LibraryPopupContentStyle}">
                 <WrapPanel>
                     <!-- Compact layout button -->
-                    <Border BorderThickness="0,0,1,0"
```
Comment (aosyatnik) [inline src/DynamoCoreWpf/Views/Search/LibraryContainerView.xaml]: I moved this background border inside of button to make background clickable.

File: src/DynamoCoreWpf/Views/Search/LibraryContainerView.xaml [hunk 2]
```diff
@@ -406,20 +406,21 @@
...
-                            </Button.Template>
-                        </Button>
-                    </Border>
+                                </Border>
+                            </ControlTemplate>
+                        </Button.Template>
+                    </Button>
 
                     <!-- Detailed layout button -->
-                    <Border Style="{StaticResource LayoutButtonStyle}">
```
Comment (aosyatnik) [inline src/DynamoCoreWpf/Views/Search/LibraryContainerView.xaml]: The same with this border.

**Comment by Racel** [comment]
@ramramps - PTAL at this PR for Search UI

File: src/DynamoCore/Search/NodeSearchModel.cs [hunk 3]
```diff
@@ -13,6 +12,33 @@ namespace Dynamo.Search
...
+        /// </summary>
+        public bool IsDetailedMode
+        {
+            get { return isDetailedMode; }
+            set
+            {
+
+                isDetailedMode = value;
+                preferences.ShowDetailedLayout = isDetailedMode;
+            }
```
Comment (ramramps) [inline src/DynamoCore/Search/NodeSearchModel.cs]: You can set the value of preferences.ShowDetailedLayout here. Get rid of private variable. Also,
 from the constructor, I see prefrences can be null. That case you can check for null and return true / false by default.

**Comment by ramramps** [comment]
Is there a test that simulates the behavior?

**Comment by aosyatnik** [comment]
Hi, @ramramps , I added tests and deleted private variable.

**Comment by aosyatnik** [comment]
Hi, @ramramps , this is great catch, I also didn't notice, that we have `dynamoViewModel` inside of `SearchViewModel`. Although it might be bad design to have dynamoVM in searchVM :)

Also I removed test. It's pretty simple code change and I don't want to create dynamoVM just for 1 test...

File: src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs [hunk 4]
```diff
@@ -149,12 +148,23 @@ public bool IsDetailedMode
...
+                {
+                    return true;
+                }
             }
             set
             {
-                isDetailedMode = value;
-                RaisePropertyChanged("IsDetailedMode");
+                if (dynamoViewModel.Model.PreferenceSettings.ShowDetailedLayout != value
+                    && dynamoViewModel.Model.PreferenceSettings != null)
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs]: NULL should be checked first.

**Comment by ramramps** [comment]
Just one change. Merge after that.

**Comment by aosyatnik** [comment]
@ramramps , I added null check. Merging it now.

**Pull request merged**